### ISSUE TITLE
Using provider from multilang resource to chain resource creation fails

### ISSUE THREAD

**Issue by viveklak** [pulumi/pulumi-eks#501]
### Problem description
See the following example:
```
"""A Kubernetes Python Pulumi program"""

import pulumi
import pulumi_eks as eks
from pulumi_kubernetes.apps.v1 import Deployment, DeploymentSpecArgs
from pulumi_kubernetes.core.v1 import (ContainerArgs, PodSpecArgs,
                                       PodTemplateSpecArgs)
from pulumi_kubernetes.meta.v1 import LabelSelectorArgs, ObjectMetaArgs

project_name = pulumi.get_project()

# Create an EKS cluster with the default configuration.
cluster1 = eks.Cluster(f"{project_name}-1")

app_labels = { "app": "nginx" }

deployment = Deployment(
    "nginx",
    spec=DeploymentSpecArgs(
        selector=LabelSelectorArgs(match_labels=app_labels),
        replicas=1,
        template=PodTemplateSpecArgs(
            metadata=ObjectMetaArgs(labels=app_labels),
            spec=PodSpecArgs(containers=[ContainerArgs(name="nginx", image="nginx")])
        ),
    ),
    __opts__ = pulumi.ResourceOptions(provider=cluster1.provider)
)

pulumi.export("name", deployment.metadata["name"])
```

Currently results in the following error:
```
Diagnostics:
  pulumi:pulumi:Stack (cluster-py-prov-dev):
    error: Program failed with an unhandled exception:
    error: Traceback (most recent call last):
      File "/opt/pulumi/bin/pulumi-language-python-exec", line 85, in <module>
        loop.run_until_complete(coro)
      File "/usr/local/Cellar/python@3.9/3.9.0_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
        return future.result()
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/stack.py", line 83, in run_in_stack
        await run_pulumi_func(lambda: Stack(func))
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/stack.py", line 51, in run_pulumi_func
        await RPC_MANAGER.rpcs.pop()
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/rpc_manager.py", line 67, in rpc_wrapper
        result = await rpc
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/resource.py", line 504, in do_register_resource_outputs
        serialized_props = await rpc.serialize_properties(outputs, {})
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 79, in serialize_properties
        result = await serialize_property(v, deps, input_transformer)
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 200, in serialize_property
        value = await serialize_property(output.future(), deps, input_transformer)
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 186, in serialize_property
        future_return = await asyncio.ensure_future(awaitable)
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/output.py", line 113, in get_value
        val = await self._future
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/output.py", line 154, in run
        value = await self._future
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/output.py", line 152, in run
        is_known = await self._is_known
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/rpc_manager.py", line 67, in rpc_wrapper
        result = await rpc
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/output.py", line 88, in is_value_known
        return await is_known and not contains_unknowns(await future)
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/output.py", line 88, in is_value_known
        return await is_known and not contains_unknowns(await future)
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/output.py", line 88, in is_value_known
        return await is_known and not contains_unknowns(await future)
      [Previous line repeated 4 more times]
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/resource.py", line 468, in do_register
        resp = await asyncio.get_event_loop().run_in_executor(None, do_rpc_call)
      File "/usr/local/Cellar/python@3.9/3.9.0_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/concurrent/futures/thread.py", line 52, in run
        result = self.fn(*self.args, **self.kwargs)
      File "/Users/vivekl/code/pulumi-eks/examples/cluster-py-prov/venv/lib/python3.9/site-packages/pulumi/runtime/resource.py", line 466, in do_rpc_call
        raise Exception(details)
    Exception: could not parse provider reference 'None::04da6b54-80e4-46f7-96ec-b56ff0331ba9': None is not a valid URN
    error: an unhandled error occurred: Program exited with non-zero exit code: 1
```

**Comment by viveklak** [comment]
CC @justinvp

**Comment by pgavlin** [comment]
> Exception: could not parse provider reference 'None::04da6b54-80e4-46f7-96ec-b56ff0331ba9': None is not a valid URN

That's odd. `None` is popping up where the provider's URN should be, and the URN should always be both known and not `None`.

**Comment by viveklak** [comment]
Verified fixed in pulumi 2.19.0 + pulumi-eks 0.22.0.
